src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java

```diff
@@ -147,7 +148,15 @@ public class LDAPConfiguration extends AbstractDescribableImpl<LDAPConfiguration
     private transient String id;
 
     @DataBoundConstructor
-    public LDAPConfiguration(@NonNull String server, String rootDN, boolean inhibitInferRootDN, String managerDN, Secret managerPasswordSecret) {
+    public LDAPConfiguration(@NonNull String server, String rootDN, boolean inhibitInferRootDN, String managerDN, Secret managerPasswordSecret){
+        if(FIPS140.useCompliantAlgorithms() && !validateServerUrlIsSecure(server)){
+            throw new IllegalArgumentException(Messages.LDAPConfiguration_InsecureServer(server));
+        }
+        String managerPassword = Secret.toString(managerPasswordSecret);
+        if(FIPS140.useCompliantAlgorithms() && StringUtils.isNotBlank(managerPassword) &&
```

Comment:
nit: for the sake of readability, maybe extracting this to a method `isPasswordCompliant` method or similar (it's used 3 times it seems)?

Reply:
Taken care